eosclubhouse/libquest.py

```diff
@@ -1536,6 +1542,18 @@ class Quest(_Quest):
     '''
 
     def _get_available(self):
```

Comment:
The logic looks good. But it would be much better to have this logic next to the dependency quests logic. I suggest you move all this logic to a `_is_contemporary_available()` internal method, then do in `_get_availability` something like:

```
    def _get_availability(self):
        # First we check if the quest should be available at this
        # datetime:
        if not self._is_contemporary_available():
            return False
        # Then we check if this quest depends on others, and if so if
        # they are complete:
        return all(self.is_named_quest_complete(q)
                   for q in self.get_dependency_quests())

```

Reply:
Done, I've implemented this using the `_get_availability` and I've also wrote a test for this functionality.